astropy/table/table.py

```diff
@@ -4025,7 +4028,166 @@ def group_by(self, keys):
+        converted to plain Column or MaskedColumn types when necessary. Time-related mixins
+        will be represented using ``np.datetime64`` or ``np.timedelta64`` as appropriate. All
+        other mixins are serialized into a flat column structure using standard representations.
+
+        Parameters
+        ----------
+        backend : str or module
+            The backend to use for conversion. This can be:
+
+            - A string, such as "pandas", "polars", or "pyarrow"
+            - The backend module itself (e.g., ``import pandas as pd; backend=pd``)
+
+        index : None, bool, str, optional
+            Specifies the index column in the resulting DataFrame.
+
+            - If None (default), use the table’s primary index if it exists and is a single column.
+            - If False, no index is set on the DataFrame.
+            - If a string, use the column with that name as the index.
+
+        use_nullable_int : bool, optional
+            If True (default), masked integer columns are converted to the backend's nullable integer type.
+            If False, an error is raised if a masked integer column is encountered.
+
+        Returns
+        -------
+        dataframe : object
+            An eager DataFrame instance as specified by the backend.
+
+        Raises
+        ------
+        ValueError
+            If the backend is not compatible with eager DataFrame conversion, or if the index argument is invalid.
+        ImportError
+            If the narwhals library is not installed.
+
+        Examples
+        --------
+        Convert a QTable with mixin columns to a pandas DataFrame:
+
+            >>> from astropy.table import QTable
+            >>> import astropy.units as u
+            >>> from astropy.time import Time, TimeDelta
+            >>> from astropy.coordinates import SkyCoord
+
+            >>> q = [1, 2] * u.m
+            >>> tm = Time([1998, 2002], format='jyear')
+            >>> sc = SkyCoord([5, 6], [7, 8], unit='deg')
+            >>> dt = TimeDelta([3, 200] * u.s)
+
+            >>> t = QTable([q, tm, sc, dt], names=['q', 'tm', 'sc', 'dt'])
+            >>> df = t.to_df(backend='pandas', index='tm')
+            >>> print(df)
+                            q  sc.ra  sc.dec              dt
+            tm
+            1998-01-01  1.0    5.0     7.0 0 days 00:00:03
+            2002-01-01  2.0    6.0     8.0 0 days 00:03:20
+
+        """
+        from ._dataframes import to_df
+
+        return to_df(
+            self, backend=backend, index=index, use_nullable_int=use_nullable_int
+        )
+
+    @classmethod
+    def from_df(
+        cls, df: Any, /, *, index: bool = False, units: Mapping[str, Any] | None = None
+    ) -> "Table":
+        """
+        Create a `~astropy.table.Table` from any ``narwhals``-compatible dataframe
+        (e.g., `pandas`, ``polars``, ``pyarrow``, etc).
+
+        Parameters
+        ----------
+        df : object
```

Comment:
careful: `Any` and `object` are actually very different as type annotations:
- `Any` represents the [top type](https://en.wikipedia.org/wiki/Any_type) (i.e., all other types are treated as valid subtype of it), or effectively, any interface is recognized as valid (even contradictory ones)
- `object` is used to mean "*no* interface can be assumed valid" (I don't know that this has a name in type theory, but that's the conventional idiom for Python type checkers).